recipes-openxt/openxt/openxt-keymanagement/key-functions

```diff
@@ -454,54 +446,88 @@ set_encrypted_key() {
 }
 
 is_xc_scheme() {
-    local root="${1:-/}"
+    local boot="${1:-/boot/system}"
     local part="${2:-/dev/xenclient/config}"
 
-    [ -e ${root}/boot/system/config/recovery-private-key.conf ] && {
+    [ -e ${boot}/config/recovery-private-key.conf ] && {
         cryptsetup luksDump ${part} | grep -q "Slot 7: ENABLED"
         return $?
     }
 
     return 1
 }
 
+recovery_unlock_key() {
+    local part="${1}"
+    local name="${2}"
+    local boot="${3}"
+    local key="${4}"
+    local extra="${5:-}"
```

Comment:
Yes, you are correct.

The ${5:-} is a pattern from when I write (bash) scripts with `set -u`.  That flag generates errors when accessing unbound variables, and numbered parameters beyond the number passed in are just such a case.  The default value provided by ${5:-} bypasses the error.